corehq/apps/reports/commtrack/standard.py

```diff
@@ -151,16 +179,37 @@ def headers(self):
     def product_data(self):
         return list(self.get_prod_data())
 
-    def get_prod_data(self):
-        sp_ids = get_relevant_supply_point_ids(self.domain, self.active_location)
-        product_ids = get_product_ids_for_program(self.domain, self.program_id) if self.program_id else None
+    def filter_by_product_ids(self):
+        """
+        This sorts by name of products that should be shown according to filter and
+        then returns the product ids to be shown according to the pagination
+        """
+        product_name_map = self._product_name_mapping
+        # sort
+        if self.request.GET.get('sSortDir_0') == 'desc':
+            sorted_product_name_map = sorted(product_name_map.items(),
+                                             key=lambda name_map: name_map[1], reverse=True)
+        else:
+            sorted_product_name_map = sorted(product_name_map.items(),
+                                             key=lambda name_map: name_map[1])
+        # product ids to filter
+        # -> all that have ledgers and
+        # -> if specific program products ids from filter
+        product_ids_to_filter = products_with_ledgers(self.domain, self._sp_ids, STOCK_SECTION_TYPE, None)
+        if self._program_product_ids:
+            product_ids_to_filter = product_ids_to_filter & set(self._program_product_ids)
+        products_to_show = [_product_name_map for _product_name_map in sorted_product_name_map
+                            if _product_name_map[0] in product_ids_to_filter]
+        # product ids according to pagination
+        return [product_id for product_id, product_name in
+                products_to_show[self.pagination.start:][:self.pagination.count]]
```

Comment:
nit: I think you could combine these last two expressions:
```python
return [
    product_id for product_id, product_name in products_to_show
    if product_id in product_ids_to_filter
][self.pagination.start:][:self.pagination.count]
```

Reply:
Beauty!